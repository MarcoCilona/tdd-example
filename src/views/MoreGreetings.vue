<template>
  <div>
    <h1 data-set-id="titleTest">{{ name }}</h1>
    <p data-set-id="paragraphGreetingsTest">Watch out, {{name}} is coming</p>
  </div>
</template>

<script lang="ts">
import { Component, Vue } from 'vue-property-decorator';

@Component
export default class MoreGreetings extends Vue {
  name = '';

  mounted() {
    this.name = this.$route.params.name;
  }
}
</script>
