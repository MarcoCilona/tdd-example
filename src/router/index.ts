import Vue from 'vue';
import VueRouter, { RouteConfig } from 'vue-router';

import Home from '@/views/Home.vue';
import MoreGreetings from '@/views/MoreGreetings.vue';

Vue.use(VueRouter);

export const routes: Array<RouteConfig> = [
  {
    path: '/',
    component: Home,
  },
  {
    path: '/more-greetings/:name',
    name: 'MoreGreetings',
    component: MoreGreetings,
  },
];

const router = new VueRouter({
  routes,
});

export default router;
