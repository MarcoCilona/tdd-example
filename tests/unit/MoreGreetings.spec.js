import { createLocalVue, mount } from '@vue/test-utils';

import VueRouter from 'vue-router';
import { routes } from '@/router/index';

// Components
import MoreGreetings from '@/views/MoreGreetings.vue';

const localVue = createLocalVue();
localVue.use(VueRouter);

/**
 * 1. Check if MoreGreetings component html is rendered correctly
 * 2. Check if More Greetings has a name data
 * 3. Check if More Greeting has a h1 element and a paragraph element
 * 4. Check if title displays name variable
 * 5. Check if paragraph contains text "Watch out, ${name} is coming"
 * 6. Check if, on created, data is equal to route name param
 */
describe('MoreGreetings tests', () => {
  let router;
  let wrapper;

  beforeEach(() => {
    router = new VueRouter({ routes, mode: 'abstract' });
    router.push('/more-greetings/Digitiamo');
    wrapper = mount(MoreGreetings, {
      localVue,
      router,
    });
  });

  it('Check MoreGreetings html', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('Check if More Greetings has a name data', async () => {
    wrapper.setData({
      name: '',
    });
    await wrapper.vm.$nextTick();
    expect(wrapper.vm.$data.name).not.toBeUndefined();
  });

  it('Check if MoreGreetings has h1 and paragraph elements', () => {
    expect(wrapper.find('[data-set-id="titleTest"]').exists()).toBeTruthy();
    expect(wrapper.find('[data-set-id="paragraphGreetingsTest"]').exists()).toBeTruthy();
  });

  it('Check if title displays name variable', async () => {
    const titleWrapper = wrapper.find('[data-set-id="titleTest"]').element;
    const testName = 'Digitiamo';
    wrapper.setData({
      name: testName,
    });

    await wrapper.vm.$nextTick();

    expect(titleWrapper.textContent).toEqual(testName);
  });

  it('Check if paragraph contains text "Watch out, name is coming"', async () => {
    const paragraphWrapper = wrapper.find('[data-set-id="paragraphGreetingsTest"]').element;
    const name = 'Digitiamo';
    wrapper.setData({
      name,
    });
    expect(paragraphWrapper.textContent).toEqual(`Watch out, ${name} is coming`);
  });

  it('Check if, on created, data is equal to route name param', async () => {
    const name = 'Digitiamo';

    await wrapper.vm.$nextTick();

    expect(wrapper.vm.$data.name).toEqual(name);
  });
});
