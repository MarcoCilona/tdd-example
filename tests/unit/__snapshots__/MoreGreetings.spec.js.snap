// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MoreGreetings tests Check MoreGreetings html 1`] = `
<div>
  <h1 data-set-id="titleTest">Digitiamo</h1>
  <p data-set-id="paragraphGreetingsTest">Watch out, Digitiamo is coming</p>
</div>
`;
